tests/nanoFramework.WebServer.Tests/WebServerTests.cs

```diff
@@ -0,0 +1,79 @@
+using nanoFramework.TestFramework;
+using System;
+
+namespace nanoFramework.WebServer.Tests
+{
+    [TestClass]
+    public class WebServerTests
+    {
+        // IsRouteMatch GET, POST, PUT, DELETE, PATCH
+        // IsRouteMatch CaseSensitive true/false
+
+        [TestMethod]
+        public void IsRouteMatch_Should_ReturnFalseForEmptyMethod()
+        {
+            // Arrange
+            var route = new CallbackRoutes()
+            {
+                Method = "GET",
+                Route = "/api/test"
+            };
+
+            // Act
+            var result = WebServer.IsRouteMatch(route, "", "/api/test");
+
+            // Assert
+            Assert.IsFalse(result);
+        }
+        
+        [TestMethod]
+        public void IsRouteMatch_Should_ReturnFalseForNotMatchingMethod()
+        {
+            // Arrange
+            var route = new CallbackRoutes()
+            {
+                Method = "GET",
+                Route = "/api/test"
+            };
+
+            // Act
+            var result = WebServer.IsRouteMatch(route, "POST", "/api/test");
+
+            // Assert
+            Assert.IsFalse(result);
+        }
+
+        [TestMethod]
+        [DataRow("GET", "/api/test", "/api/test")]
+        [DataRow("POST", "/api/test", "/api/test")]
+        [DataRow("PUT", "/api/test", "/api/test")]
+        [DataRow("PATCH", "/api/test", "/api/test")]
+        [DataRow("DELETE", "/api/test", "/api/test")]
+        [DataRow("GET", "/API/TEST", "/api/test")]
+        [DataRow("POST", "/API/TEST", "/api/test")]
+        [DataRow("PUT", "/API/TEST", "/api/test")]
+        [DataRow("PATCH", "/API/TEST", "/api/test")]
+        [DataRow("DELETE", "/API/TEST", "/api/test")]
+        [DataRow("GET", "/api/test", "/API/TEST")]
+        [DataRow("POST", "/api/test", "/api/test")]
+        [DataRow("PUT", "/api/test", "/API/TEST")]
+        [DataRow("PATCH", "/api/test", "/API/TEST")]
+        [DataRow("DELETE", "/api/test", "/API/TEST")]
```

Comment:
Can you add couple of longer URL with parameters just to see how it does react?

Reply:
Sure, I'll add few more. If you have any edge cases to check, leave me a note and I'll add more tests.

Reply:
I think testing withn things like this can be great:
* /api/test?id=1234
* /api/test/?id=1234
* /api/test?
* /api/api/test
* /api/api/test?
* /api/api/test/?id=1234
* /api/api/test?id=1234
* /tagada/ap/test

things like this!

Reply:
Is the "/api/api/test/?id=1234" valid URL? (/ and then ?)
I'm asking because current implementation fails to resolve it.

Reply:
Added few more tests for routes.
Later I'll add few more.

Reply:
I think that's all for now. I'll extend tests when working on improving web server (better tests would be - create controller , so the "callbackroute" is created based on that and the check if route matches)

Reply:
> Is the "/api/api/test/?id=1234" valid URL? (/ and then ?)
> I'm asking because current implementation fails to resolve it.

Well, never tested that one and I asked Bing chat for some help.

"Yes, that is a valid URL. The characters / and ? can be together like that, as they have different meanings in different parts of the URL.

The / character is used to separate path segments in the URL. In your example, the path is /api/api/test/, which means that the resource you are requesting is located under the test directory, which is under the api directory, which is under the api directory, which is under the root directory of the server.

The ? character is used to separate the query string from the rest of the URL. The query string is a set of key-value pairs that provide additional information to the server. In your example, the query string is id=1234, which means that you are passing the value 1234 for the key id to the server.

The query string is optional and can be omitted if you don’t need to pass any information to the server. However, the / character is mandatory and cannot be omitted, as it indicates the end of the path and the start of the query string.

If you want to learn more about the syntax and semantics of URLs, you can check out these resources:

[Web Naming and Addressing Overview (URIs, URLs, …)](https://www.w3.org/Addressing/)
[Uniform Resource Locators - World Wide Web Consortium (W3C)](https://www.w3.org/Addressing/URL/url-spec.html)
[RFC 1738 - Uniform Resource Locators (URL) - IETF Datatracker](https://datatracker.ietf.org/doc/html/rfc1738)
I hope this helps. If you have any other questions, please let me know. 😊"

So, we may have to treat it as a valid URL. For the time being, we can ignore it and add it to the list of improvements. It's definitely a edge case!